mgr/dashboard: new generic HTTP error page component (#36715)
ceph/ceph: src/pybind/mgr/dashboard/HACKING.rst

To overwrite or extend the default CSS, you can add your own styles in
``./frontend/src/vendor.overrides.scss``.

Error Handling
~~~~~~~~~~~~~~

> For handling front-end erros, there is a generic Error Component which can be

Reviewer: ```suggestion
For handling front-end errors, there is a generic Error Component which can be
```

found in ``./src/pybind/mgr/dashboard/frontend/src/app/core/error``. For 
reporting a new error, you can simply extend the ``DashboardError`` class

> in 'error.ts' file and add specific header and message for the new error. Some

Reviewer: Keep in mind that this document uses reStructuredText markup, not Markdown:
```suggestion
in ``error.ts`` file and add specific header and message for the new error. Some
```

generic error classes are already in place such as 'DashboardNotFoundError' and

> 'DashboardForbiddenError' which can be called and reused in different scenarios.

Reviewer: ```suggestion
``DashboardForbiddenError`` which can be called and reused in different scenarios.
```
